src/Silex/ViewListenerWrapper.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/*
+ * This file is part of the Silex framework.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Silex;
+
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Event\GetResponseForControllerResultEvent;
+
+/**
+ * Wraps view listeners.
+ *
+ * @author Dave Marshall <dave@atst.io>
+ */
+class ViewListenerWrapper
+{
+    protected $app;
+    protected $callback;
+
+    /**
+     * Constructor.
+     *
+     * @param Application $app      An Application instance
+     * @param callable    $callback
+     */
+    public function __construct(Application $app, $callback)
+    {
+        $this->app = $app;
+        $this->callback = $callback;
+    }
+
+    public function __invoke(GetResponseForControllerResultEvent $event)
+    {
+        $controllerResult = $event->getControllerResult();
+        $callback = $this->app['callback_resolver']->resolveCallback($this->callback);
+
+        if (!$this->shouldRun($callback, $controllerResult)) {
+            return;
+        }
+
+        $response = call_user_func($callback, $controllerResult, $event->getRequest());
+
+        if ($response instanceof Response) {
+            $event->setResponse($response);
+        } else {
+            $event->setControllerResult($response);
```

Comment:
I don't see why it should be undocumented. It becomes effectively a pipeline anyway, just with skippable steps.  That's the same as Symfony's View events, where if you don't ->setControllerResult() then that's the equivalent of returning null.

(The alternative if you really want to pipeline it with lots of object conversion is to leverage http://www.github.com/Crell/transformer, recently released! :smile: )